File: ICSharpCode.Decompiler.Tests/TestCases/VBPretty/VBClosures.vb
Imports System
Imports System.Collections.Generic

Public Class VBClosures
	Public Shared Function PerIteration(ByVal items As IEnumerable(Of Integer), ByVal factor As Integer) As List(Of Func(Of Integer))
		Dim result As New List(Of Func(Of Integer))()
		For Each item As Integer In items
			Dim local As Integer = item * factor
			result.Add(Function() local + factor)
		Next
		Return result
	End Function

	Public Shared Function SingleCapture(ByVal factor As Integer) As Func(Of Integer, Integer)
		Return Function(n) n * factor
	End Function
End Class

File: ICSharpCode.Decompiler.Tests/TestCases/VBPretty/VBClosures.cs
using System;
using System.Collections.Generic;

public class VBClosures
{
	public static List<Func<int>> PerIteration(IEnumerable<int> items, int factor)
	{
		List<Func<int>> list = new List<Func<int>>();
		checked
		{
			foreach (int item in items)
			{
				int num = item * factor;
				list.Add(() => num + factor);
			}
			return list;
		}
	}

	public static Func<int, int> SingleCapture(int factor)
	{
		return (int n) => checked(n * factor);
	}
}
